temporal-sdk/src/main/java/io/temporal/worker/tuning/SlotSupplier.java

```diff
@@ -0,0 +1,93 @@
+ *     {@link WorkflowSlotInfo}, {@link ActivitySlotInfo}, and {@link LocalActivitySlotInfo}.
+ */
+@Experimental
+public interface SlotSupplier<SI extends SlotInfo> {
+  /**
+   * This function is called before polling for new tasks. Your implementation should block until a
+   * slot is available, then return a permit to use that slot.
+   *
+   * @param ctx The context for slot reservation.
+   * @return A permit to use the slot which may be populated with your own data.
+   * @throws InterruptedException The worker may choose to interrupt the thread in order to cancel
+   *     the reservation, or during shutdown. You may perform cleanup, and then should rethrow the
+   *     exception.
+   */
+  SlotPermit reserveSlot(SlotReserveContext<SI> ctx) throws InterruptedException;
+
+  /**
+   * This function is called when trying to reserve slots for "eager" workflow and activity tasks.
+   * Eager tasks are those which are returned as a result of completing a workflow task, rather than
+   * from polling. Your implementation must not block, and If a slot is available, return a permit
+   * to use that slot.
+   *
+   * @param ctx The context for slot reservation.
+   * @return Maybe a permit to use the slot which may be populated with your own data.
+   */
+  Optional<SlotPermit> tryReserveSlot(SlotReserveContext<SI> ctx);
+
+  /**
+   * This function is called once a slot is actually being used to process some task, which may be
+   * some time after the slot was reserved originally. For example, if there is no work for a
+   * worker, a number of slots equal to the number of active pollers may already be reserved, but
+   * none of them are being used yet. This call should be non-blocking.
+   *
+   * @param ctx The context for marking a slot as used.
+   */
+  void markSlotUsed(SlotMarkUsedContext<SI> ctx);
+
+  /**
+   * This function is called once a permit is no longer needed. This could be because the task has
+   * finished, whether successfully or not, or because the slot was no longer needed (ex: the number
+   * of active pollers decreased). This call should be non-blocking.
+   *
+   * @param ctx The context for releasing a slot.
+   */
+  void releaseSlot(SlotReleaseContext<SI> ctx);
+
+  /**
+   * Because we currently use thread pools to execute tasks, there must be *some* defined
+   * upper-limit on the size of the thread pool for each kind of task. You must not hand out more
+   * permits than this number. If unspecified, the default is {@link Integer#MAX_VALUE}. Be aware
+   * that if your implementation hands out unreasonable numbers of permits, you could easily
+   * oversubscribe the worker, and cause it to run out of resources.
+   *
+   * <p>This value should never change during the lifetime of the supplier.
+   *
+   * @return the maximum number of slots that can ever be in use at one type for this slot type.
+   */
+  default int maximumSlots() {
```

Comment:
```suggestion
  default int getMaximumSlots() {
```

Pedantic and don't have strong opinion here, just saw https://github.com/temporalio/sdk-java/pull/2014#discussion_r1555820733 marked resolved

Reply:
Fixed, must have marked it by accident.